document-store/src/test/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutorTest.java

```diff
@@ -516,6 +522,72 @@ public void testFindWithKeyAndRelationalFilter() throws IOException, URISyntaxEx
     verify(iterable).cursor();
   }
 
+  @Test
+  public void testOptimizeSorts_simpleAggregate() throws IOException, URISyntaxException {
+    final Query query =
+        Query.builder()
+            .addSelection(IdentifierExpression.of("item"))
+            .addSort(IdentifierExpression.of("item"), DESC)
+            .build();
+
+    testAggregation(query, "mongo/pipeline/optimize_sorts_simple_aggregate.json");
+  }
+
+  @Test
+  public void testOptimizeSorts_sortSpecNotInSelection() throws IOException, URISyntaxException {
+    final Query query =
+        Query.builder()
+            .addSelection(IdentifierExpression.of("item"))
+            .addSort(IdentifierExpression.of("item.description"), DESC)
+            .build();
+
+    testAggregation(query, "mongo/pipeline/optimize_sorts_sort_spec_not_in_selection.json");
+  }
+
+  @Test
+  public void testOptimizeSorts_sortSpecNotInSelection_selectionWithAlias()
+      throws IOException, URISyntaxException {
+    final Query query =
```

Comment:
Can you also add a test for different alias names for projections.

E.g.:
`SELECT item.name AS name FROM ... ORDER BY name ASC`.

Ideally, we should first sort by `item.name` and then project it with the alias `name`.